you-should-use.plugin.zsh

```diff
@@ -60,18 +60,22 @@ function _check_ysu_hardcore() {
 function _check_git_aliases() {
   if [[ "$1" = "git "* ]]; then
       local found=false
-      local tokens
+      local cmd
       local k
       local v
-      git config --get-regexp "^alias\..+$" | sort | while read entry; do
-        tokens=("${(@s/ /)entry}")
-        k="${tokens[1]#alias.}"
-        v="${tokens[2]}"
-
-        if [[ "$2" = "git $v" || "$2" = "git $v "* ]]; then
-          ysu_message "git alias" "$v" "git $k"
-          found=true
-        fi
+      git config --get-regexp "^alias\..+$" | sort | while read k v; do
+        k="${k#alias.}"
+        cmd=( ${=2##git} )
+
+        for word in ${=v}; do
+          echo "$word in $cmd:  ${cmd[(ie)$word]} -le ${#cmd}"
+          if ! [[ ${cmd[(ie)$word]} -le ${#cmd} ]]; then
+            continue 2 # at least one word in alias not in command
```

Comment:
what is the 2 after the continue for?

Reply:
there are two nested loops here, the 2 causes `continue` to affect the outer loop and look at the next alias instead of only the next word in the current one

Reply:
I was not actually aware you could do this - I guess  I learnt something new today :)